src/Bonus.java

```diff
@@ -1,66 +1,102 @@
-import java.util.HashSet;
-import java.util.Objects;
-import java.util.Scanner;
+import java.util.*;
 
 public class Bonus {
 
-    public static void main(String[] args) {
+  public static void main(String[] args) {
 
-        // Onderstaande code is uitgecommentarieerd. Haal de code uit de comment (delete /* en */) om aan de slag te gaan met de bonus
-        // In de code staan op verschillende plekken comments tussen "/*...*/" als hint wat je moet doen
+    // Onderstaande code is uitgecommentarieerd. Haal de code uit de comment (delete /* en */) om aan de slag te gaan met de bonus
+    // In de code staan op verschillende plekken comments tussen "/*...*/" als hint wat je moet doen
 
+    // The first method is being used to create a random number that will saved in a
+    HashSet<Integer> secretnumber = randomnumbergenerator();
+    // HastSet collection.
+    // This will force the numbers that are being saved to be unique.
+    // This is still an array that need to be looped over
+    String stringnumber = setToStringConverter(
+      secretnumber); // THis method will add all values of the HashSet Collection array to be add to one string
 
-        HashSet<Integer> secretnumber = randomnumbergenerator();
-        String stringnumber = setToStringConverter(secretnumber);
-        System.out.println(stringnumber);
-        feedback(/*vul hier het juiste argument in*/);
+    System.out.println(stringnumber);
+    // Now the fun can begin, we can try to gues the number
+    feedback(stringnumber);
 
-    }
+  }
 
-    public static void/*moet dit returntype "void" zijn of wat anders?*/ randomnumbergenerator(/*Heeft deze methode nog parameter(s) nodig?*/){
+  public static HashSet<Integer> randomnumbergenerator() {
          /*
         Vul hier de body van de methode in.
-
         Stappenplan:
         - Maak een nieuwe variabele van type Random. (Tip: Zoek op internet hoe je Random kunt gebruiken)
         - Maak een nieuwe variabele van type HashSet.
         - Schrijf een while-loop om 4 random nummers aan de hashset toe te voegen
         - return de hashset
          */
+
+    Random rand = new Random();
+    HashSet<Integer> numbers = new HashSet<>();
+    while (numbers.size() < 4) {
```

Comment:
dit is wel mooi trucje

Reply:
Yup! een makkelijk manier om de loop door te lopen en deze te verhogen met door `.add`.